tcms/xmlrpc/tests/test_testrun.py

```diff
@@ -202,3 +203,39 @@ def test_create_with_product_version(self):
         # the result is still using product_version from TR.plan.product_version
         # not the one we specified above
         self.assertEqual(result['product_version'], self.plan.product_version.value)
+
+
+class TestUpdateTestRun(XmlrpcAPIBaseTest):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.initial_test_plan = TestPlanFactory()
+        self.initial_build = BuildFactory()
+        self.initial_summary = 'Initial summary.'
+        self.test_run = TestRunFactory(
+            plan=self.initial_test_plan,
+            build=self.initial_build,
+            summary=self.initial_summary,
+        )
+
+        self.updated_test_plan = TestPlanFactory()
+        self.updated_build = BuildFactory()
+        self.updated_summary = 'Updated summary.'
+        self.updated_stop_date = '2020-05-05'
+
+    def test_successful_update(self):
+        update_fields = {
+            'plan': self.updated_test_plan.pk,
+            'build': self.updated_build.pk,
+            'summary': self.updated_summary,
+            'stop_date': self.updated_stop_date
+        }
+
+        self.rpc_client.exec.TestRun.update(self.test_run.pk, update_fields)
+        updated_test_run = TestRun.objects.get(pk=self.test_run.pk)
```

Comment:
Use `self.test_run.refresh_from_db()` to minimize the number of intermediate variables that are used. This also takes away some of the naming confusion.

Bonus tip: `exec.TestRun.update()` will return a dict of the updated values. You can use that to compare against the values in DB and against the values from RPC params. This will assert not only that you are updating the object in DB but that the RPC call is returning the updated values as well.

Reply:
Done. I have added more assertion, using the method you suggested